### ISSUE TITLE
Getting warnings even with allowTypedFunctionExpressions set to true

### ISSUE THREAD

**Issue by saranshkataria** [typescript-eslint/typescript-eslint#384]
After adding the
`rules: { allowTypedFunctionExpressions: true }`
for the alpha release (#149), I still see them as warnings

![image](https://user-images.githubusercontent.com/1450819/55121720-fbc97800-50b8-11e9-825f-8a2e549b348a.png)

**Repro**

Code repo: https://github.com/saranshkataria/frontend-starter-kit

```JSON
{
  rules: {
    allowTypedFunctionExpressions: true,
  }
}
```

```TS
import React from 'react';
import { hot } from 'react-hot-loader/root';

const App: React.FunctionComponent<{}> = () => <div>Hello World</div>;
App.displayName = 'App';
export default hot(App);
```

**Expected Result**
No warnings

**Actual Result**
Get warnings

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.5.1-alpha.5` |
| `@typescript-eslint/parser`        | `1.5.1-alpha.5` |
| `TypeScript`                       | `3.3.4000` |
| `ESLint`                           | `5.15.3` |
| `node`                             | `8.15.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
In this instance, you have eslint misconfigured.
The correct syntax for configuring rules is:
```
{
  "rules": {
    "<rule_name>": ["<level>", <options>],
  }
}
```

I.e. for `@typescript-eslint/explicit-function-return-type`:
```json
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["error", {
      "allowTypedFunctionExpressions": true
    }],
  }
}
```

**Comment by saranshkataria** [comment]
Thanks!

**Comment by saranshkataria** [comment]
@bradzacher I made the changes and it works for tsx files, but when I run lint, it fails on test files. Since I already have the types included, it shouldn't?

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-function-return-type.md#allowexpressions

**Comment by saranshkataria** [comment]
I have that enabled as well, but I still see an error

**Comment by saranshkataria** [comment]
I put in core-js3 in my babel config and the issue got resolved. Not sure why though.

**Comment by ButterHub** [comment]
Should @typescript-eslint not default to enabling allowTypedFunctionExpressions. Seems like everyone would face this issue otherwise? Am i overlooking something?

**Comment by bradzacher** [comment]
It's going to be the default in v2. There is a PR waiting to be merged (the last one before the v2 release)

**Comment by 0v1se** [comment]
Is it the same issue?
Or is it intended behavior?

`
        const values = ["test"];
        values.map(v => v);
`
it's highlighted even if "allowTypedFunctionExpressions": true
